src/state/apis/nft/nftApi.ts

```diff
@@ -28,26 +29,39 @@ export const nftApi = createApi({
   endpoints: build => ({
     getNftUserTokens: build.query<NftItem[], GetNftUserTokensInput>({
       queryFn: async ({ accountIds }, { dispatch }) => {
-        const sources = [
-          zapperApi.endpoints.getZapperNftUserTokens,
-          covalentApi.endpoints.getCovalentNftUserTokens,
+        const services = [
+          getAlchemyNftData,
+          (accountIds: AccountId[]) =>
+            dispatch(zapperApi.endpoints.getZapperNftUserTokens.initiate({ accountIds })),
+          (accountIds: AccountId[]) =>
+            dispatch(covalentApi.endpoints.getCovalentNftUserTokens.initiate({ accountIds })),
         ]
 
-        const results = await Promise.all(
-          sources.map(source => dispatch(source.initiate({ accountIds }))),
-        )
-        const data = results.reduce<NftItem[]>((acc, result) => {
+        const results = await Promise.all(services.map(service => service(accountIds)))
+
+        const data = results.reduce<Record<AssetId, NftItem>>((acc, result) => {
           if (result.data) {
             const { data } = result
-            acc = acc.concat(data)
+
+            data.forEach(item => {
+              const nftAssetId: AssetId = `${item.id}-${item.collection.id}`
```

Comment:
Manual construction from the collection CAIP-19 and nft ID is removed two PRs on top of this one once we start normalizing and bring `assetId` in as a property
https://github.com/shapeshift/web/pull/4596/files#diff-7ef9f6fd70f7be4faed5cc8c2702f7d46529689cf81962a71d88be8c46a23025R102